src/test/java/seedu/address/testutil/TypicalEvents.java

```diff
@@ -59,7 +59,8 @@ private TypicalEvents() {} // prevents instantiation
      * Returns a list of typical events.
      */
     public static List<Event> getTypicalEvents() {
-        return new ArrayList<>(Arrays.asList(TECH_CONFERENCE, ART_EXHIBITION, SPORTS_FESTIVAL));
+        return new ArrayList<>(Arrays.asList(new Event(TECH_CONFERENCE),
+                new Event(ART_EXHIBITION), new Event(SPORTS_FESTIVAL)));
```

Comment:
Made this function use defensive copies of the events instead, as some of the changes to these events were affecting the behavior of the other test cases